### PR TITLE
Add support for Private Link

### PR DESCRIPTION
**Pull request by anrudraw**
Link to approved swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/11193

### CODE REVIEW

File: sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/Resource.cs [hunk 1]
```diff
@@ -63,75 +55,25 @@ public Resource(Kind kind, string location, string id = default(string), string
...
-        /// Gets the resource type.
+        /// Gets the type of the resource. E.g.
+        /// "Microsoft.Compute/virtualMachines" or
+        /// "Microsoft.Storage/storageAccounts"
         /// </summary>
         [JsonProperty(PropertyName = "type")]
         public string Type { get; private set; }
 
-        /// <summary>
-        /// Gets or sets the kind of the service. Possible values include:
```
Comment (markcowl) [inline sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/Resource.cs:84]: This seems like a breaking change

File: sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/ServicesDescription.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ namespace Microsoft.Azure.Management.HealthcareApis.Models
     /// <summary>
     /// The description of the service.
     /// </summary>
-    public partial class ServicesDescription : Resource
+    public partial class ServicesDescription : ServicesResource
```
Comment (markcowl) [inline sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/ServicesDescription.cs:21]: Additional breaking change

File: sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/ServicesResourceIdentity.cs [hunk 3]
```diff
@@ -17,26 +17,26 @@ namespace Microsoft.Azure.Management.HealthcareApis.Models
     /// Setting indicating whether the service has a managed identity
     /// associated with it.
     /// </summary>
-    public partial class ResourceIdentity
+    public partial class ServicesResourceIdentity
```
Comment (markcowl) [inline sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/ServicesResourceIdentity.cs:20]: breaking change

**Review by markcowl — CHANGES_REQUESTED** [review]
It looks like this involves breaking changes to your Swagger, which prodiced breaking changes in your client.  It also looks like the build is failing on tests, perhaps due to soem of the changes.  You'll need to fix these..

Is this breaking change intentional?  Generally, breaking changes in the API require approval from the API review board

File: sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/Resource.cs (hunk 1 above)
Comment (anrudraw) [reply sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/Resource.cs:84]: @markcowl this changes is coming from the common-types referenced by the Private Link. "$ref": "../../../../../common-types/resource-management/v1/privatelinks.json#/definitions/PrivateEndpointConnectionListResult". That's the reason I renamed the model definition in swagger spec from Resources -> ServiceResource.

File: sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/ServicesDescription.cs (hunk 2 above)
Comment (anrudraw) [reply sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/ServicesDescription.cs:21]: This changes is approved by the ARM team. Only model definition has been renamed here from Resource -> ServicesResource.

File: sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/Resource.cs (hunk 1 above)
Comment (anrudraw) [reply sdk/healthcareapis/Microsoft.Azure.Management.HealthcareApis/src/Generated/Models/Resource.cs:84]: link to the PR approved by ARM team: https://github.com/Azure/azure-rest-api-specs-pr/pull/2046

**Comment by anrudraw** [comment]
@markcowl I have fixed the failing tests. All checks have passed.

**Review by markcowl — APPROVED** [review]

**Pull request merged**